src/UI/UIView.cpp

```diff
@@ -860,6 +969,45 @@ void UIView::save_trans_functions()
   }
 }
 
+void UIView::save_user_session()
+{
+  std::string fileName = m_dir_to_save + helper::get_OS_path_separator() + m_save_file_name + ".usr";
+  std::ofstream savefile(fileName);
+  
+  if (savefile.is_open())
+  {
+    if (!m_controller_app.get_loaded_file().empty())
+    {
+      savefile << "volume_loaded " << m_controller_app.get_loaded_file() << "\n";
+    }
+    savefile << "alpha_multiplier " << std::to_string(m_multiplier) << "\n";
+    savefile << "threshold " << std::to_string(m_threshold) << "\n";
+    savefile << "scale " << std::to_string(m_scale) << "\n";
+    savefile << "z-scale " << std::to_string(m_z_scale) << "\n";
+    savefile << "Slices " << std::to_string(m_slices) << "\n";
+    savefile << "automatic _slice adjustment " << std::to_string(m_dynamic_slices) << "\n";
+    savefile << "RenderMethod " << std::to_string(m_rendermethod) << "\n";
+    savefile << "Render_Channel " << std::to_string(m_renderchannel) << "\n";
+    savefile << "Render_Volume_data " << std::to_string(m_renderVolume) << "\n";
+
+    savefile << "ClipXmin " << std::to_string(m_clip_min.x) << "\n";
+    savefile << "ClipXmax " << std::to_string(m_clip_max.x) << "\n";
+    savefile << "ClipYmin " << std::to_string(m_clip_min.y) << "\n";
+    savefile << "ClipYmax " << std::to_string(m_clip_max.y) << "\n";
+    savefile << "ClipZmin " << std::to_string(m_clip_min.z) << "\n";
+    savefile << "ClipZmax " << std::to_string(m_clip_max.z) << "\n";
+    savefile << "Use_transferfunction " << std::to_string(m_use_transferfunction) << "\n";
+    
+    if (m_use_transferfunction)
+    {
+      save_trans_functions();
```

Comment:
saving the transfer functions separately is nice, but could also maybe open up to some weird bugs where the transfer functions get deleted and then the session restored, but now the transfer function isn't there.  Would it be safer to save the transfer functions in the session file? @mirestrepo @eldu what do you think?

Reply:
Is there any utility that we get for saving them in a separate file?

Saving the transfer functions in the session file would help minimize the number of files we're saving to and we wouldn't need to find the transfer function file for the session file since it would be in it. I like it!